### ISSUE TITLE
Term 2023 issues

### ISSUE THREAD

**Issue by mmaelicke** [data-hydenv/data#18]
@maxschmi for reference

This is a report on errors, warnings and issues logged by the server while uploading this years data.

Referencing @data-hydenv/students for your information. If you recognize some of the information below, you are welcome to comment, otherwise just ignore the mentioning.

### metadata
I guess this is a student who dropped out.
```
Some entries are missing metadata:
----------------------------------
            name  hobo_id github_name data_available region  latitude  longitude exposition altitude influence description
14  ***, ***      NaN         NaN            NaN    NaN       NaN        NaN        NaN      NaN       NaN         NaN
```

Additionally, there was an exception, which means there might be missing metadata

```
[ERROR]: (psycopg2.errors.UndefinedFunction) operator does not exist: character varying = timestamp without time zone
LINE 3: ...RE details.key = 'altitude' AND details.str_value = '1900-01...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.

[SQL: SELECT details.id AS details_id, details.key AS details_key, details.str_value AS details_str_value, details.int_value AS details_int_value, details.float_value AS details_float_value, details.bool_value AS details_bool_value
FROM details
WHERE details.key = %(key_1)s AND details.str_value = %(str_value_1)s
 LIMIT %(param_1)s]
[parameters: {'key_1': 'altitude', 'str_value_1': datetime.datetime(1900, 1, 2, 12, 0), 'param_1': 1}]
(Background on this error at: https://sqlalche.me/e/14/f405)
```
This basically means one student had a date in the *altitude* column, which was parsed as the 2.1.1900 12:00. Maybe putted `2.1` into the cell and Excel formatted as a date?

### Raw data
These are either extra HOBO time series and can be ignored, or the Metadata file references the wrong HOBO (typo?)
```
File ./data-master/hobo/2023/raw/10350089.csv references HOBO ID=10350089, which is not found.
File ./data-master/hobo/2023/raw/10347324.csv references HOBO ID=10347324, which is not found.
File ./data-master/hobo/2023/raw/10347327.csv references HOBO ID=10347327, which is not found.
File ./data-master/hobo/2023/raw/10347333.csv references HOBO ID=10347333, which is not found.
File ./data-master/hobo/2023/raw/10350107.csv references HOBO ID=10350107, which is not found.
File ./data-master/hobo/2023/raw/10347366.csv references HOBO ID=10347366, which is not found.
```

**Comment by MarilenaAlmeida** [comment]
The value in the altitude column was supposed to be 2. Unfortunately I can no longer access the file to correct this.

**Comment by mmaelicke** [comment]
> The value in the altitude column was supposed to be 2. Unfortunately I can no longer access the file to correct this.

Thanks for reporting. I corrected this.

**Comment by maxschmi** [comment]
Hey Mirko,
Yes their are additional HObos that are not in the Meta table. Those were my spare Hobos I did install as a backup.

**Comment by mmaelicke** [comment]
@data-hydenv/students to let you all know, I have a minor issue uploading the quality-checked data. I will solve that during the day and then the quality checked data will be available in the database, as well.

**Comment by mmaelicke** [comment]
**@data-hydenv/students  the 2023 quality checked data is uploaded.**

There were some warnings and errors, read through them and reply if you feel like you know something about it.

```
File ./data-master/hobo/2023/hourly/10347324_Th.csv references HOBO ID=10347324, which is not found.

Parsing file './data-master/hobo/2023/hourly/10347355_Th.csv' was not successfull.
Do not edit the files by hand!
Error: 'DataFrame' object has no attribute 'tstamp' 

File ./data-master/hobo/2023/hourly/10350089_Th.csv references HOBO ID=10350089, which is not found.

Parsing file './data-master/hobo/2023/hourly/10349996_Th.csv' was not successfull.
Do not edit the files by hand!
Error: time data '2022.12.01. 00:00:00' does not match format '%Y.%m.%d %H:%M:%S' 

File ./data-master/hobo/2023/hourly/10347333_Th.csv references HOBO ID=10347333, which is not found.

Parsing file './data-master/hobo/2023/hourly/10350049_Th.csv' was not successfull.
Do not edit the files by hand!
Error: 'DataFrame' object has no attribute 'tstamp' 

File ./data-master/hobo/2023/hourly/10350107_Th.csv references HOBO ID=10350107, which is not found.
```

The error `Error: 'DataFrame' object has no attribute 'tstamp' ` indicates, that reading the `ddtm` column failed. If it was your file, please check it carefully.


**Respond to this thread, if you changed something about your data**

**Comment by arrpit19** [comment]
Oh, alright I'll make the changes.
object has no attribute 'tstamp.
Could you please point me in the right direction? As how to do it?
Thanks.

**Comment by kg200** [comment]
Hello Mirko, 
concerning:  File ./data-master/hobo/2023/hourly/10350089_Th.csv references HOBO ID=10350089, which is not found.
just changed my Hobo ID in the metafile yesterday because I had to use a spare one and forgot to change it until now. I hope it works now. 
all the best Kilian     Am Freitag, 20. Januar 2023 um 10:47:48 MEZ hat ARPIT KUMAR ***@***.***> Folgendes geschrieben:  
 
 

Oh, alright I'll make the changes.
object has no attribute 'tstamp.
Could you please point me in the right direction? As how to do it?
Thanks.

**Comment by nico51two** [comment]
Hi, the missing timestamp was most likely due to me having the wrong 
column name for the dttm column. Fixed and reuploaded. Sorry :)

**Comment by adia-saurey** [comment]
Hello,
The ref_id and ref_r2 columns for hobo id 10350095 have been added.
Thank you!

**Comment by MarilenaAlmeida** [comment]
Hi, 
I've corrected and re-uploaded the data from id 10347333.
I hope now it works!

**Comment by arrpit19** [comment]
Hi, I have added all the information for 10350107.  Hope it works this time
:)

**Comment by mmaelicke** [comment]
@arrpit19, @MarilenaAlmeida, @nico51two, @kg200 thanks for your quick replies!

We produced a new error :)

```
[ERROR]: File: './data-master/hobo/2023/hourly/10350049_Th.csv' errored. Message: (psycopg2.errors.DatatypeMismatch) column "value" is of type real but expression is of type text
LINE 1: ...ity_flag_id) SELECT meta_id, variable_id, tstamp, value, qua...
                                                             ^
HINT:  You will need to rewrite or cast the expression.

[SQL: INSERT INTO data (meta_id, variable_id, tstamp, value, quality_flag_id) SELECT meta_id, variable_id, tstamp, value, quality_flag_id FROM hnnwjjpjmcizpgpz ON CONFLICT (meta_id, variable_id, tstamp) DO NOTHING]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

it is the file: './data-master/hobo/2023/hourly/10350049_Th.csv', please check the temperature column. I guess there is some kind of character there, as it presents itself as a character datatype.

**Comment by nico51two** [comment]
yeah sorry I only now managed to work my backlog from being sick in the first week so I guess my file still had NAs in there... i'm finishing up as we speak and will upload in ~5 mins so check again in 15 :D ...sorry for the inconvenience

**Comment by nico51two** [comment]
Aaaaaaand it's up...thanks for your patience

**Comment by xamfru** [comment]
@mmaelicke 

My uploaded file was read in correctly, when i was trying to aggregate for calculating indices of all HOBOS in 2023 i saw that there are only 38 unique timestamps in my file. So i uploaded it with corrected timestamps. Is it possible to update the db data table once again?

Sorry for the inconvience

**Comment by mmaelicke** [comment]
> @mmaelicke
> 
> My uploaded file was read in correctly, when i was trying to aggregate for calculating indices of all HOBOS in 2023 i saw that there are only 38 unique timestamps in my file. So i uploaded it with corrected timestamps. Is it possible to update the db data table once again?
> 
> Sorry for the inconvience

Thanks for correcting the file. I will try to update everything this evening, but everyone's really busy with the database right now, and I don't want to interrupt. If your data in the database is a bit messed up, don't worry and continue with the version online, that's alright.

**Comment by kg200** [comment]
@mmaelicke 
I still can't find my HOBO in the metadata. I changed the ID about a week ago, because I had to use one of Max's HOBOs. 
Did you also update the metadata table? If so I'll probably need to do some changes again.

**Comment by kg200** [comment]
Hey group! 
Did anyone succed to have different icons for the different terms and a color coding for the mean temperatures at the same time in plot_ly? 
Would be happy if you someone coud help me there!! 
Thank you! And have a nice weekend to all that are ready with their report :)
